### PR TITLE
fix:click on skip button in Peer Review after orientation change then app is crashed

### LINKED ISSUE #4143: click on skip button in Peer Review after orientation change then app is crashed
**Summary:** 

**issue :  when  click on skip button in Peer Review after  orientation change  then app is crashed** 

**Step to Reproduce the issue**

1. Open the Peer Review Activity 
2. change the orientation of screen 
3. click on the button "skip this image"
4. then App crash

**Screenshot for first issue **

**System logs:**
2020-12-31 18:14:01.518 26369-26369/fr.free.nrw.commons.beta E/ACRA: ACRA caught a NullPointerException for fr.free.nrw.commons.beta
    java.lang.NullPointerException: Attempt to invoke virtual method 'void android.widget.Button.setEnabled(boolean)' on a null object reference**
        at fr.free.nrw.commons.review.ReviewImageFragment.disableButtons(ReviewImageFragment.java:199)
        at fr.free.nrw.commons.review.ReviewPagerAdapter.disableButtons(ReviewPagerAdapter.java:51)
        at fr.free.nrw.commons.review.ReviewActivity.lambda$onCreate$0$ReviewActivity(ReviewActivity.java:116)
        at fr.free.nrw.commons.review.-$$Lambda$ReviewActivity$-LATUATvQAPhlK652ow3LysvMck.onClick(Unknown Source:2)
        at android.view.View.performClick(View.java:6608)
        at android.view.View.performClickInternal(View.java:6585)
        at android.view.View.access$3100(View.java:785)
        at android.view.View$PerformClick.run(View.java:25919)
        at android.os.Handler.handleCallback(Handler.java:873)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:201)
        at android.app.ActivityThread.main(ActivityThread.java:6810)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:547)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:873)

**Device and Android version:** 
Android Device:Redmi y3;
API :29

**Commons app version:** 
2.13.1 or current master branch 
  
**Would you like to work on the issue?**
YES

Please let us know whether you want to fix the issue by yourself. If not, anyone can get the issue assigned to them.
Yes

### PR DESCRIPTION
**Pull request by Prince-kushwaha**
**Description (required)**

Fixes #4143 

What changes did you make and why?

Issue is due to calling   methods of ReviewImageFragment from ReviewActivity is wrong
So i add the other way to call the ReviewImageFragment methods  from ReviewActivity 
it solve the issue

**Screen-shots**

https://user-images.githubusercontent.com/65972015/105719903-d45bc480-5f48-11eb-8bb7-3957d032af14.mp4

  
**Tests performed (required)**
**build varient-prodebug and betadebug on device Android:Redmi y3**

### CODE REVIEW

**Comment by Prince-kushwaha** [comment]
@nicolas-raoul  review my PR

**Comment by Prince-kushwaha** [comment]
@misaochan @madhurgupta10 review my PR

**Comment by neslihanturan** [comment]
Hey @Prince-kushwaha , please add screen cast to display this changes solve the issue.

**Comment by Prince-kushwaha** [comment]
@neslihanturan i add screen shot

**Comment by Prince-kushwaha** [comment]
@nicolas-raoul @neslihanturan please review my this PR

File: app/src/main/java/fr/free/nrw/commons/review/ReviewActivity.java [hunk 1]
```diff
@@ -55,6 +55,7 @@
     ReviewHelper reviewHelper;
     @Inject
     DeleteHelper deleteHelper;
+    private ReviewImageFragment reviewImageFragment;
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/review/ReviewActivity.java:58]: Please add javadoc

File: app/src/main/java/fr/free/nrw/commons/review/ReviewActivity.java [hunk 2]
```diff
@@ -98,7 +99,7 @@ protected void onCreate(Bundle savedInstanceState) {
 
         reviewPagerAdapter = new ReviewPagerAdapter(getSupportFragmentManager());
         reviewPager.setAdapter(reviewPagerAdapter);
-        reviewPagerAdapter.getItem(0);
+        reviewPager.setOffscreenPageLimit(2);
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/review/ReviewActivity.java:102]: Please add a comment explaining why the number `2`.

File: app/src/main/java/fr/free/nrw/commons/review/ReviewActivity.java [hunk 3]
```diff
@@ -113,7 +114,8 @@ protected void onCreate(Bundle savedInstanceState) {
         }
 
         btnSkipImage.setOnClickListener(view -> {
-            reviewPagerAdapter.disableButtons();
+            reviewImageFragment=getInstanceOfReviewImageFragment();
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/review/ReviewActivity.java:117]: Please a space before and after `=`.
Here and in other similar lines of your PR.

Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/review/ReviewActivity.java:117]: Please a space before and after `=`.
(here and in other similar lines of your PR)

File: app/src/main/java/fr/free/nrw/commons/review/ReviewActivity.java [hunk 4]
```diff
@@ -226,4 +230,10 @@ public boolean onOptionsItemSelected(MenuItem item) {
         }
         return super.onOptionsItemSelected(item);
     }
+
+    public ReviewImageFragment getInstanceOfReviewImageFragment(){
+        int pos=reviewPager.getCurrentItem();
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/review/ReviewActivity.java:235]: In Java we prefer to use full words for variable names, unlike other languages like C.

File: app/src/main/java/fr/free/nrw/commons/review/ReviewActivity.java [hunk 5]
```diff
@@ -226,4 +230,10 @@ public boolean onOptionsItemSelected(MenuItem item) {
...
+        reviewImageFragment= (ReviewImageFragment) reviewPagerAdapter.instantiateItem(reviewPager,pos);
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/review/ReviewActivity.java:236]: Please add a space after each `,`

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]
I confirm that the crash:
- still happens in master
- is fixed by this PR

File: app/src/main/java/fr/free/nrw/commons/review/ReviewActivity.java [hunk 6]
```diff
@@ -226,4 +229,10 @@ public boolean onOptionsItemSelected(MenuItem item) {
         }
         return super.onOptionsItemSelected(item);
     }
+
+    public ReviewImageFragment getInstanceOfReviewImageFragment(){
+        int currentItemOfReviewPager = reviewPager.getCurrentItem();
+        reviewImageFragment = (ReviewImageFragment) reviewPagerAdapter.instantiateItem(reviewPager,currentItemOfReviewPager);
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/review/ReviewActivity.java:235]: space after `,`

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]

File: app/src/main/java/fr/free/nrw/commons/review/ReviewActivity.java (hunk 1 above)
Comment (nicolas-raoul) [reply app/src/main/java/fr/free/nrw/commons/review/ReviewActivity.java:58]: Any difficulty to address this? Feel free to take your time though :-)

File: app/src/main/java/fr/free/nrw/commons/review/ReviewActivity.java [hunk 7]
```diff
@@ -226,4 +233,10 @@ public boolean onOptionsItemSelected(MenuItem item) {
         }
         return super.onOptionsItemSelected(item);
     }
+
+    public ReviewImageFragment getInstanceOfReviewImageFragment(){
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/review/ReviewActivity.java:237]: Sorry, javadoc needed here too

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]

**Review by nicolas-raoul — APPROVED** [review]

**Pull request merged**